models/spleen_deepedit_annotation/configs/inference.json

```diff
@@ -97,13 +98,48 @@
                 "keys": "image",
                 "label_names": "@label_names",
                 "number_intensity_ch": "@number_intensity_ch"
+            }
+        ],
+        "deepedit_transforms": [
+            {
+                "_target_": "AddGuidanceFromPointsDeepEditd",
+                "ref_image": "image",
+                "guidance": "guidance",
+                "label_names": "@label_names"
+            },
+            {
+                "_target_": "Resized",
+                "keys": "image",
+                "spatial_size": "@spatial_size",
+                "mode": "area"
             },
+            {
+                "_target_": "ResizeGuidanceMultipleLabelDeepEditd",
+                "guidance": "guidance",
+                "ref_image": "image"
+            },
+            {
+                "_target_": "AddGuidanceSignalDeepEditd",
+                "keys": "image",
+                "guidance": "guidance",
+                "number_intensity_ch": "@number_intensity_ch"
+            }
+        ],
+        "extra_transforms": [
             {
                 "_target_": "EnsureTyped",
                 "keys": "image"
             }
         ]
     },
+    "preprocessing": {
+        "_target_": "Compose",
+        "transforms": "$@infer#preprocessing_transforms + @infer#auto_seg_transforms + @infer#extra_transforms"
+    },
```

Comment:
Hi @diazandr3s , I think we can simply add one flag to control whether to use deepedit preprocessing and we may not need the "infer" section.
For example, we can define:
```
"use_deepedit": false,
```
then this section can be:
```
    "preprocessing": {
        "_target_": "Compose",
        "transforms": "$@preprocessing_transforms + (@deepedit_transforms if use_deepedit else @auto_seg_transforms) + @extra_transforms"
    },
```
and we will also not need to define `deepedit_postprocessing`

Reply:
Hi @yiheng-wang-nv,

Thanks for checking this out and for your feedback.
DeepEdit pre_transforms are a bit complicated. Here are the pre_transforms and the logic: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/infers/deepedit.py#L75-L107

Also, here are the infer classes we need to define depending on the model type (segmentation/depeedit): https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/configs/deepedit.py#L143-L164

Is it possible to implement this complexity as you are suggesting with a single flag? How to implement a conditional within the inference json file.

Let me know,

Reply:
Thanks again @yiheng-wang-nv.

I'd also like to simplify things, but it is not only defining the preprocessing. Post transforms are subject to the type of preprocessing done. Post transforms should be applied according to the applied preprocessing transforms.

In addition to this, there are some pre-transforms that are cached for faster inference: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/infers/deepedit.py#L83C13-L83C13

We may also want to consider this for fast inference

Reply:
Hi @diazandr3s , sorry I still cannot understand your point here.
In the monailabel link you posted, it also just use one flag `self.type == InferType.DEEPEDIT` to determine which set of transforms should be extended, so why can't we use the similar way in the bundle?
I do not see any blocker here, could you give an example?

Or maybe we can have a quick meeting to discuss it? cc @tangy5 @Nic-Ma